mlrun/datastore/targets.py

```diff
@@ -891,6 +891,16 @@ def add_writer_step(
             max_events=self.max_events,
             flush_after_seconds=self.flush_after_seconds,
             update_last_written=featureset_status.update_last_written_for_target,
+            dict_fields=[
+                "path",
+                "columns",
+                "index_cols",
+                "partition_cols",
+                "time_field",
+                "storage_options",
+                "max_events",
+                "flush_after_seconds",
+            ],  # update_last_written is not serializable, so we must exclude it (ML-5108)
```

Comment:
maybe we can build this list once and store into the class? would also help in keeping this list in sync with the __init__() param list (as will be closer together)
even nicer if you can come with some python meta trick to create this list automatically)

Reply:
Well, the meta trick is what it does if you don't pass it the list 😅. I'll see what I can do.

What I'd really want to do here is not to serialize the writer step at all (because why should be? It's derived from the class – e.g. `storey.ParuqetTarget` from `ParquetTarget`). But I'm afraid that's a much bigger and riskier change to make.